web_src/js/components/RepoActionView.vue

```diff
@@ -222,8 +228,8 @@ const sfc = {
       return await resp.json();
     },
 
-    async loadJob() {
-      if (this.loading) return;
+    async loadData({force = false} = {}) {
+      if (this.loading && !force) return;
```

Comment:
In the event that a force request is received, I think that there will be a race condition that may cause a similar problem to the one that you're attempting to solve, occasionally:

There's loadData(A) which was already running, and then loadData(B) which was running with the force flag.  If loadData(B) returns from the server first, and then loadData(A)'s data returns from the server afterwards, I believe the log data will be *reverted* back to the data from the first request.

Reply:
Yup, that's why I blocked this PR from merging https://github.com/go-gitea/gitea/pull/31124#issuecomment-2191810381